docs/SECURE_PLUGIN_COMMUNICATION.md

```diff
@@ -0,0 +1,124 @@
+<!--
+http://www.apache.org/licenses/LICENSE-2.0.txt
+
+
+Copyright 2017 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# Secure Plugin Communication
+
+Snap communicates with plugins over gRPC protocol, which in general transfers data in plaintext.
+Snap allows securing communication with plugins by opening TLS channels and using certificates to authenticate plugins and framework.
+
+## Usage 
+
+This walkthrough assumes you have downloaded a Snap release as described in [Getting Started](../README.md#getting-started).
+
+### Shortest guide
+
+Assuming all the test files are available, the following steps will result in secure plugin communication:
+
+```
+snapteld --log-level 1 --plugin-trust 0 --tls-cert /tmp/snaptest-cli.crt --tls-key /tmp/snaptest-cli.key --ca-cert-paths /tmp/snaptest-ca.crt
+## (in another terminal)
+## Load each plugin
+snaptel plugin load --plugin-cert /tmp/snaptest-srv.crt --plugin-key /tmp/snaptest-srv.key --plugin-ca-certs /tmp/snaptest-ca.crt plugins/snap-plugin-collector-rand
+## Start a sample task
+snaptel task create -t sample-task.json
+```
+
+### Detailed preparation
+
+Starting secure communication requires following steps:
+1. Obtain TLS certificate and private key for framework.
+    * Please note that this certificate should allow usage for TLS web client authentication (as specified in RFC 3280)
+1. Obtain TLS certificate and private key for each plugin or group of plugins.
+    * Please note that this certificate should allow usage for TLS web server authentication (as specified in RFC 3280)
+1. Obtain and locate the CA certificates that are necessary to authenticate framework and plugin certificates.
+
+Process of acquiring a TLS certificate is a complex one. Every organization has its specific rules on security, thus the details are not given here.
+
+We do provide a short guide on obtaining self-signed certificates that may be used for tests outside production environment; see [Obtaining self-signed TLS certificates for tests](#obtaining-self-signed-tls-certificates-for-tests).
+
+### Enabling secure communication
+
+Secure communication is enabled by passing the required paths to programs: `snapteld`, and plugin (via `snaptel`). The minimum paths necessary are:
```

Comment:
It is possible. However REST API documentation needs update to put such information. Opened issue #1629 to address this.